src/traces/candlestick/attributes.js

```diff
@@ -0,0 +1,49 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var OHLCattrs = require('../ohlc/attributes');
+var boxAttrs = require('../box/attributes');
+
+var directionAttrs = {
+    visible: {
```

Comment:
in https://github.com/plotly/plotly.js/pull/1020/commits I chose to remove `increasing` / `decreasing` `visible` and add `increasing` / `decreasing` `name` and `showlegend` so that users can edit legend item name / visibility per direction, but disallowing hiding a direction completely for the graph.